presto-product-tests/src/main/java/io/prestosql/tests/hive/TestCsv.java

```diff
@@ -13,24 +13,76 @@
  */
 package io.prestosql.tests.hive;
 
+import com.google.inject.Inject;
+import io.prestosql.tempto.AfterTestWithContext;
+import io.prestosql.tempto.BeforeTestWithContext;
 import io.prestosql.tempto.ProductTest;
 import io.prestosql.tempto.assertions.QueryAssert.Row;
+import io.prestosql.tempto.hadoop.hdfs.HdfsClient;
 import io.prestosql.tempto.query.QueryResult;
 import org.testng.annotations.Test;
 
+import java.io.ByteArrayInputStream;
 import java.util.List;
 
 import static com.google.common.collect.ImmutableList.toImmutableList;
 import static io.prestosql.tempto.assertions.QueryAssert.Row.row;
 import static io.prestosql.tempto.assertions.QueryAssert.assertThat;
 import static io.prestosql.tempto.query.QueryExecutor.query;
 import static io.prestosql.tests.TestGroups.STORAGE_FORMATS;
+import static io.prestosql.tests.utils.QueryExecutors.onHive;
+import static io.prestosql.tests.utils.QueryExecutors.onPresto;
 import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
 public class TestCsv
         extends ProductTest
 {
     private static final String TPCH_SCHEMA = "tiny";
+    private static final String HEADER = "###HEADER###\n";
+    private static final String FOOTER = "\n###FOOTER###";
+    private static final String DATA = "value1,value2,value3";
+
+
+    @Inject
+    private HdfsClient hdfsClient;
+
+    @BeforeTestWithContext
```

Comment:
This executes before each test.

The best would be to use test requirements here. So you could define `HiveTableDefinition` that would automatically handle data upload. See `io.prestosql.tests.hive.TestHivePartitionsTable#getRequirements` and usage of `io.prestosql.tempto.fulfillment.table.hive.InlineDataSource#createStringDataSource`

That way you don't need to clean this up, because tempto will take care of this.

Also IIRC tables that are defined in `resources/sql-tests/datasets/` should be also available here via `ImmutableTablesState` so you would need to setup anything here.